packages/concerto-core/lib/introspect/stringvalidator.js

```diff
@@ -60,14 +89,38 @@ class StringValidator extends Validator{
      */
     validate(identifier, value) {
         if(value !== null) {
-            if(!this.regex.test(value)) {
-                this.reportError(identifier, 'Value \'' + value + '\' failed to match validation regex: ' + this.regex);
+            if (this.regex) {
+                if(!this.regex.test(value)) {
+                    this.reportError(identifier, 'Value \'' + value + '\' failed to match validation regex: ' + this.regex);
+                }
+            }
+            //Enforce string length rule after regex check
```

Comment:
This comment suggests that length checking should come first.
https://github.com/accordproject/concerto/issues/643#issuecomment-1542134421

Reply:
I can update the logic, but I don't think order of execution is gonna solve the problem if regex has length constraint in it

Reply:
You're right, it won't. We just have the design principle to apply the least complex validation first.